sql-plugin/src/main/spark311/scala/com/nvidia/spark/rapids/shims/AnsiCastShim.scala

```diff
@@ -46,4 +47,12 @@ object AnsiCastShim {
       m.getBoolean(e)
     case _ => false
   }
+
+  def getEvalMode(c: Cast): GpuEvalMode.Value = {
```

Comment:
nit: is there any way that we could try to look at `ansiEnabled`instead of going off of the config. It is possible for a cast to bypass the default parameter, which is the config, and provide a different value. At least in some versions of Spark.